lib/iris/tests/unit/common/lenient/test_Lenient.py

```diff
@@ -4,179 +4,166 @@
-    def setUp(self):
-        self.lenient = Lenient()
+class Test___getitem__:
+    def test_in(self, lenient):
+        assert lenient["maths"] is True
 
-    def test_in(self):
-        self.assertTrue(self.lenient["maths"])
-
-    def test_not_in(self):
+    def test_not_in(self, lenient):
         emsg = "Invalid .* option, got 'MATHS'."
-        with self.assertRaisesRegex(KeyError, emsg):
-            _ = self.lenient["MATHS"]
-
+        with pytest.raises(KeyError, match=emsg):
+            _ = lenient["MATHS"]
 
-class Test___repr__(tests.IrisTest):
-    def setUp(self):
-        self.lenient = Lenient()
 
-    def test(self):
+class Test___repr__:
+    def test(self, lenient):
         expected = "Lenient(maths=True)"
-        self.assertEqual(expected, repr(self.lenient))
+        assert expected == repr(lenient)
 
 
-class Test___setitem__(tests.IrisTest):
-    def setUp(self):
-        self.lenient = Lenient()
-
-    def test_key_invalid(self):
+class Test___setitem__:
+    def test_key_invalid(self, lenient):
         emsg = "Invalid .* option, got 'MATHS."
-        with self.assertRaisesRegex(KeyError, emsg):
-            self.lenient["MATHS"] = False
+        with pytest.raises(KeyError, match=emsg):
+            lenient["MATHS"] = False
 
-    def test_maths_value_invalid(self):
-        value = sentinel.value
+    def test_maths_value_invalid(self, mocker, lenient):
+        value = mocker.sentinel.value
         emsg = f"Invalid .* option 'maths' value, got {value!r}."
-        with self.assertRaisesRegex(ValueError, emsg):
-            self.lenient["maths"] = value
+        with pytest.raises(ValueError, match=emsg):
+            lenient["maths"] = value
 
-    def test_maths_disable__lenient_enable_true(self):
-        self.assertTrue(_LENIENT.enable)
-        self.lenient["maths"] = False
-        self.assertFalse(self.lenient.__dict__["maths"])
-        self.assertFalse(_LENIENT.enable)
+    def test_maths_disable__lenient_enable_true(self, lenient):
+        assert _LENIENT.enable is True
+        lenient["maths"] = False
+        assert lenient.__dict__["maths"] is False
+        assert _LENIENT.enable is False
 
-    def test_maths_disable__lenient_enable_false(self):
+    def test_maths_disable__lenient_enable_false(self, lenient):
         _LENIENT.__dict__["enable"] = False
-        self.assertFalse(_LENIENT.enable)
-        self.lenient["maths"] = False
-        self.assertFalse(self.lenient.__dict__["maths"])
-        self.assertFalse(_LENIENT.enable)
-
-    def test_maths_enable__lenient_enable_true(self):
-        self.assertTrue(_LENIENT.enable)
-        self.lenient["maths"] = True
-        self.assertTrue(self.lenient.__dict__["maths"])
-        self.assertTrue(_LENIENT.enable)
-
-    def test_maths_enable__lenient_enable_false(self):
+        assert _LENIENT.enable is False
+        lenient["maths"] = False
+        assert lenient.__dict__["maths"] is False
+        assert _LENIENT.enable is False
```

Comment:
Are you happy that the testing for the exact values False/True here is the best expression of what was intended ?

I.E. I assume you want to ensure that the value is actually a "True", and not a 1 or "x" or anything else for which "if x" or bool(x) counts true. And similar for the "is False" tests.

If that is **_not_** so, then I think the style here is not correct : we should just be using "assert x" / "assert not x" instead.

The problem is, I think in _unittest_ language you would commonly do assertTrue/assertFalse for "assert x"/"assert not x" anyway.
But if it _really_ meant the same as what is here, that should probably be an "assertIs".

Again, that applies to a whole lot of cases here !